lib/jobs/dell-wsman-getComponent.js

```diff
@@ -0,0 +1,107 @@
+// Copyright © 2017 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+'use strict';
+
+var di = require('di');
+module.exports = WsmanGetComponentFactory;
+di.annotate(WsmanGetComponentFactory, new di.Provide('Job.Dell.Wsman.getComponent'));
+di.annotate(WsmanGetComponentFactory, new di.Inject(
+    'Job.Dell.Wsman.Base',
+    'JobUtils.WsmanTool',
+    'Logger',
+    'Assert',
+    'Util',
+    'Services.Configuration',
+    'Errors',
+    'Services.Encryption',
+    'validator'
+));
+
+function WsmanGetComponentFactory(
+    BaseJob,
+    WsmanTool,
+    Logger,
+    assert,
+    util,
+    configuration,
+    errors,
+    encryption,
+    validator
+) {
+    var logger = Logger.initialize(WsmanGetComponentFactory);
+
+    function WsmanGetComponentJob(options, context, taskId) {
+        WsmanGetComponentJob.super_.call(this, logger, options, context, taskId);
+        assert.object(this.options);
+        this.nodeId = this.context.target;
+    }
+
+    util.inherits(WsmanGetComponentJob, BaseJob);
+
+    WsmanGetComponentJob.prototype._initJob = function () {
+        var self = this;
+        self.dell = configuration.get('dell');
+        if (!self.dell || !self.dell.services || !self.dell.services.configuration) {
+            throw new errors.NotFoundError('Dell SCP  web service is not defined in smiConfig.json.');
+        }
+    };
+
+    WsmanGetComponentJob.prototype._handleSyncRequest = function() {
+        var self = this;
+        return self.checkOBM('SCP create xml for deleting volume')
+        .then(function(obm){
+            return self.getComponent(obm);
+        });
+    };
+
+    WsmanGetComponentJob.prototype._handleSyncResponse = function(response) {
+        var json = JSON.parse(response.body);
+        logger.info('Status from SCP Microservice for getComponent:  ' + json["status"]);
+        if (json["status"] === "OK") {
+            return response;
+        } else {
+            throw new Error("Failed to getComponent from smi service");
+        }
+    };
+
+    WsmanGetComponentJob.prototype.getComponent = function(obm) {
+        if (!validator.isIP(obm.config.host)) {
+            throw new Error('Invalid ServerIP');
```

Comment:
if you design the function to return a promise, then you'd better also wrap the error in promise, you will find it will become easier to write unit-test.